src/test/crypto.cc

```diff
@@ -115,6 +346,42 @@ TEST(AES, Decrypt) {
   delete kh;
 }
 
+TEST(AES, DecryptNoBl) {
+  CryptoHandler *h = g_ceph_context->get_crypto_handler(CEPH_CRYPTO_AES);
+  const char secret_s[] = {
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+    0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+  };
+  bufferptr secret(secret_s, sizeof(secret_s));
+
+  const unsigned char ciphertext[] = {
+    0xb3, 0x8f, 0x5b, 0xc9, 0x35, 0x4c, 0xf8, 0xc6,
+    0x13, 0x15, 0x66, 0x6f, 0x37, 0xd7, 0x79, 0x3a,
+    0x11, 0x90, 0x7b, 0xe9, 0xd8, 0x3c, 0x35, 0x70,
+    0x58, 0x7b, 0x97, 0x9b, 0x03, 0xd2, 0xa5, 0x01,
+  };
+
+  const unsigned char want_plaintext[] = {
+    0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+    0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff,
+  };
+  unsigned char plaintext[sizeof(want_plaintext)];
+
+  std::string error;
+  CryptoKeyHandler *kh = h->get_key_handler(secret, error);
```

Comment:
nit, use `unique_ptr<>` for holding `kh`. otherwise static analyzer won't complain potential mem leak. as `ASSERT_EQ()` will bail out if it fails without freeing `kh`.

Reply:
Switched to `std::unique_ptr`.